bootloader/src/pyi_utils.c

```diff
@@ -1028,159 +1057,279 @@ pyi_utils_create_child(const char *thisfile, const ARCHIVE_STATUS* status,
+                            if (path) {
+                                ok = CFStringGetFileSystemRepresentation(path, buf, sizeof(buf));
+                                CFRelease(path); /* free */
+                            }
+                            CFRelease(url); /* free */
+                            if (!ok) {
+                                VS("LOADER [AppleEvent ARGV_EMU]: "
+                                   "Failed to convert file:/// path to POSIX filesystem representation for arg %ld!\n",
+                                   index);
+                                continue;
+                            }
+                        }
+                    }
+                    /* Append URL to argv_pyi array, reallocating as necessary */
+                    VS("LOADER [AppleEvent ARGV_EMU]: arg[%d] = %s\n", (int)argc_pyi, buf);
+                    tmp_str = strdup(buf);
+                    tmp_argv = (char **)realloc(argv_pyi, (argc_pyi + 2) * sizeof(char *));
+                    if (!tmp_argv || !tmp_str) {
+                        /* Out of memory. Extremely unlikely -- not clear what to do here.
+                         * Attempt to silently continue. */
+                        VS("LOADER [AppleEvent ARGV_EMU]: allocation for arg[%d] failed: %s\n",
+                           argc_pyi, strerror(errno));
+                        free(tmp_argv); /* free of NULL ok */
+                        free(tmp_str);
+                        continue;
+                    }
+                    argv_pyi = tmp_argv;
+                    argv_pyi[argc_pyi++] = tmp_str;
+                    argv_pyi[argc_pyi] = NULL;
+                    VS("LOADER [AppleEvent ARGV_EMU]: argv entry appended.\n");
+                }
+            }
 
-     argv_pyi = (char**)realloc(argv_pyi,(argc_pyi+2)*sizeof(char*));
-     argv_pyi[argc_pyi++] = strdup(buffer);
-     argv_pyi[argc_pyi] = NULL;
+            err = AEDisposeDesc(&docList);
 
-     VS("LOADER [ARGV_EMU]: argv entry appended.");
+            return err;
 
-     free(buffer);
-   }
+        } else {
+            /* The child process exists.. so we forward events to it */
 
-  err = AEDisposeDesc(&docList);
+            ProcessSerialNumber psn;
+            AppleEvent evtCopy;
+            AEAddressDesc target;
+            OSStatus err = noErr;
+            AEEventClass evtClass;
+            AEEventID evtID;
 
+            if (apple_event_is_open_uri) {
+                evtClass = kInternetEventClass;
+                evtID = kAEGetURL;
+            } else {
+                /* open doc */
+                evtClass = kCoreEventClass;
+                evtID = kAEOpenDocuments;
+            }
 
-  return (err);
+            VS("LOADER [AppleEvent EVT_FWD]: Will forward %s event to child pid %ld...\n", descStr, (long)child_pid);
+
+            err = GetProcessForPID(child_pid, &psn);
+            if (err != noErr) return (OSErr)err;
```

Comment:
Could you prevent those short lines please?
```suggestion
            if (err != noErr) {
                return (OSErr)err;
            }
```

Reply:
Do we have to do this?  It adds visual noise to have curly braces and 3 lines for what is a boilerplate error check that happens like 5 times in that function (and other places). 

FWIW, this is idiomatic in C -- I realize to Python sensibilities it is offensive but to C sensibilities it's not necessarily so.

 Let me know if you insist I will do this.

Reply:
We do not have C conventions actually. I do not insist :)

Reply:
Ok, thanks.  It would make the code just noisier IMHO -- I only did this for boilerplace error checks. Thanks.

Reply:
For what it's worth, I prefer the oneliners... just do whatever suits you.